src/client/Client.h

```diff
@@ -1414,6 +1415,9 @@ class Client : public Dispatcher, public md_config_obs_t {
 
   uint64_t cap_hits = 0;
   uint64_t cap_misses = 0;
+
+  ceph::spinlock delay_lock;
+  std::multimap<Inode*,int> delay_release;
```

Comment:
In case, for example, if there has two threads will put the same `inode` almost at the same time or before the `delay_release` was cleared by `tick()`, the same inode is put again, without `multimap` the second one will just replace the first one, then we will hit the ref leak. Locally I could reproduce it easily.

Without the `multimap`, it also will work if the `inode` has already in the map and then just to increase the count. But this will make the code a little compicated IMO.

Reply:
delay_release[in] += n; should work

Reply:
Yeah, it works, I will fix it. Thanks.